framework/include/utils/MooseMeshUtils.h

```diff
@@ -289,4 +290,182 @@ void makeOrderedNodeList(std::vector<std::pair<dof_id_type, dof_id_type>> & node
                          std::vector<dof_id_type> & elem_id_list,
                          std::vector<dof_id_type> & ordered_node_list,
                          std::vector<dof_id_type> & ordered_elem_id_list);
+
+/**
+ * Split a HEX8 element into five TET4 element to prepare for the cutting operation.
+ * @param mesh The mesh to be modified
+ * @param elem_id The id of the element to be split
+ * @param converted_elems_ids a vector to record the ids of the newly created TET4 elements
+ */
+void hexElemSplitter(ReplicatedMesh & mesh,
+                     const dof_id_type elem_id,
+                     std::vector<dof_id_type> & converted_elems_ids);
+
+/**
+ * Split a PYRAMID5 element into two TET4 element to prepare for the cutting operation.
+ * @param mesh The mesh to be modified
+ * @param elem_id The id of the element to be split
+ * @param converted_elems_ids a vector to record the ids of the newly created TET4 elements
+ */
+void pyramidElemSplitter(ReplicatedMesh & mesh,
+                         const dof_id_type elem_id,
+                         std::vector<dof_id_type> & converted_elems_ids);
+
+/**
+ * Split a PRISM6 element into three TET4 element to prepare for the cutting ope ration.
+ * @param mesh The mesh to be modified
+ * @param elem_id The id of the element to be split
+ * @param converted_elems_ids a vector to record the ids of the newly created TET4 elements
+ */
+void prismElemSplitter(ReplicatedMesh & mesh,
```

Comment:
do we absolutely need the mesh here? Or can we pass a reference to the element instead of mesh + id